object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionTransfer.java

```diff
@@ -74,9 +88,54 @@ public ObjectConstructionTransfer(final CFAnalysis analysis) {
             result.getElseStore(),
             exceptionalStores);
     exceptionalStores = null;
+
+    Node receiver = node.getTarget().getReceiver();
+    addTemporaryVars(result, node);
+    if (atypefactory.biMap.inverse().containsKey(receiver)) {
+      String methodName = node.getTarget().getMethod().getSimpleName().toString();
+      methodName =
+          ((CalledMethodsAnnotatedTypeFactory) atypeFactory)
+              .adjustMethodNameUsingValueChecker(methodName, node.getTree());
+      accumulate(atypefactory.biMap.inverse().get(receiver), result, methodName);
+    }
+
     return finalResult;
   }
 
+  @Override
+  public TransferResult<CFValue, CFStore> visitObjectCreation(
+      ObjectCreationNode node, TransferInput<CFValue, CFStore> input) {
+    TransferResult<CFValue, CFStore> result = super.visitObjectCreation(node, input);
+    addTemporaryVars(result, node);
+    return result;
+  }
+
+  private void addTemporaryVars(TransferResult<CFValue, CFStore> result, Node node) {
+    if (atypefactory.hasMustCall(node.getTree())) {
+      if (atypefactory.hasMustCall(node.getTree())) {
+
+        LocalVariableNode localVariableNode;
+        if (!atypefactory.biMap.inverse().containsKey(node)) {
+          VariableTree temp = createTemporaryVar(node);
+          IdentifierTree identifierTree = treeBuilder.buildVariableUse(temp);
+          localVariableNode = new LocalVariableNode(identifierTree);
+          localVariableNode.setInSource(true);
+          atypefactory.biMap.put(localVariableNode, node);
+        } else {
+          localVariableNode = atypefactory.biMap.inverse().get(node);
+        }
```

Comment:
I would extract this code into its own method, `getOrCreateTempVar`

Reply:
https://github.com/kelloggm/object-construction-checker/pull/291/files#diff-3fc5dcd77eafbbf61a8da8156fac14c917b05d258d829aab8e8706b815b91b52R148